dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.41/V2_41_2__Create_table_custom_icon.sql

```diff
@@ -1,22 +1,28 @@
+
+
+-- customicon table creation
+
 CREATE TABLE IF NOT EXISTS customicon
 (
-    customiconid   int8 GENERATED ALWAYS AS IDENTITY,
-    "key"          varchar(100) NOT NULL,
-    fileresourceid int8         NOT NULL,
-    description    text         NULL,
-    keywords       text[]       NULL,
-    created        TIMESTAMP    NULL DEFAULT CURRENT_TIMESTAMP,
-    createdby      bigint       NOT NULL,
-    CONSTRAINT customicon_pkey PRIMARY KEY (customiconid),
-    CONSTRAINT customicon_ukey UNIQUE ("key"),
+    id                  bigint         NOT NULL,
+    uid                 character      varying(11) NOT NULL,
+    code                character      varying(50),
+    iconkey             varchar(100)   NOT NULL,
+    fileresourceid      bigint         NULL,
```

Comment:
Shouldn't all icons have a file resource?

Reply:
Yes all need file but customicon needs a reference to that external file via fileResource. Whereas bundled/default icon does not need that.